### DISCUSSION TITLE
Klipper Screen (makerpase klipad v1.2) upgrade

### DISCUSSION THREAD

**Discussion by Roshi9** [bassamanator/Sovol-SV06-firmware#172] [Q&A]
Hi, i have a sv06 with sovol klipper screen, board inside says makerbase klipad 50 v1.2 which is a remix of the v1.1 judging from pictures i found online. I was wondering if anyone was able to upgrade from factory firmware to real klipper. Switching to a different board or pc just to get latest klipper and kamp is a last resort i would like to avoid.

**Comment by bassamanator** [comment]
> kamp is a last resort i would like to avoid

Not sure if I follow.

~~Please~~People have upgraded their sovol screens. I think the user Rory has created a `README` on how to backup or restore the screen to factory settings if something goes wrong.

**Comment by DEVisions** [comment]
In case someone is still looking for the same information:
- https://github.com/torte71/armbian-mksklipad50
- https://torte71.github.io/InsideSovolKlipperScreen

I can confirm I successfully upgraded to bookworm and latest Klipper. If upgrading on a SV06 for the printer firmware you can use the one provided by @bassamanator in the misc folder of his repo.
